### PR TITLE
Add begin() and rollback() methods to Distributed Transaction API

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds `begin()` and `rollback()` methods to Distributed Transaction API. I'll add inline comments for the details. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java [hunk 1]
```diff
@@ -165,13 +187,24 @@ DistributedTransaction start(String txId, Isolation isolation, SerializableStrat
...
 
   /**
-   * Aborts a given transaction.
+   * Rolls back a given transaction.
+   *
+   * @param txId a transaction ID
+   * @return {@link TransactionState}
+   * @throws TransactionException if aborting the given transaction failed
+   */
+  TransactionState rollback(String txId) throws TransactionException;
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java:196]: Also added `rollback()` to `DistributedTransactionManager`.

File: core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java [hunk 2]
```diff
@@ -165,13 +187,24 @@ DistributedTransaction start(String txId, Isolation isolation, SerializableStrat
...
+
+  /**
+   * Aborts a given transaction. This method is an alias of {@link #rollback(String)}.
    *
    * @param txId a transaction ID
    * @return {@link TransactionState}
    * @throws TransactionException if aborting the given transaction failed
    */
-  TransactionState abort(String txId) throws TransactionException;
+  default TransactionState abort(String txId) throws TransactionException {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java:205]: After this change, this `abort()` will be regarded as an alias of `rollback()`.

File: rpc/src/main/proto/scalardb.proto [hunk 3]
```diff
@@ -309,6 +309,8 @@ service DistributedTransaction {
   }
   rpc GetState(GetTransactionStateRequest) returns (GetTransactionStateResponse) {
   }
+  rpc Rollback(RollbackRequest) returns (RollbackResponse) {
```
Comment (brfrn169) [inline rpc/src/main/proto/scalardb.proto:312]: Also added the sane methods (RPCs) to proto.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Comment by Torch3333** [comment]
@brfrn169 Sorry, I don't think you requested my review yet. Please ignore it.

**Comment by brfrn169** [comment]
> @brfrn169 Sorry, I don't think you requested my review yet. Please ignore it.

@feeblefakie Thank you for reviewing it. I just requested your review. Thanks.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**